src/components/SwapWidget/LimitOrder/index.tsx

```diff
@@ -312,7 +335,7 @@ const LimitOrder: React.FC<Props> = ({ swapType, setSwapType, isLimitOrderVisibl
   }, [attemptingTxn, showConfirm, swapErrorMessage, trade, txHash]);
 
   const onCurrencySelect = useCallback(
-    (currency) => {
+    (currency: any, tokenDrawerType?: LimitNewField) => {
```

Comment:
There is issue here now. tokenDrawerType is defined at 2 places. one as state variable and one as method argument. As now you have added to method argument, logic inside method will always take value from argument and if its not passed it will take as undefined, it will never consider state value inside method. make sure to use different name in method argument. and inside method give priority to method argument then state variable like this e.g. `const type = fromArg ?? fromState` then use that type everywhere